### PR TITLE
Test Utils

### PR DESCRIPTION
**Pull request by xingdi-eric-yuan**

### CODE REVIEW

File: froggy/tests/test_utils.py [hunk 1]
```diff
@@ -93,3 +110,498 @@ def token_counter(text):
...
+    config_contents["cot"] = {
+        "random_seed": 43,
+        "max_steps": 50,
+        "cot_style": "standard",
+        "llm_name": "gpt20",
+        "llm_temperature": [0.3, 0.5],
+    }
+
+    # write the config file into yaml
+    with open(config_file, "w") as f:
```
Comment (chisingh) [inline froggy/tests/test_utils.py:391]: May be we should mock file open instead. Something like the following:
@patch('builtins.open', new_callable=mock_open, read_data="key: value\nnested:\n  key: value")

File: froggy/tests/test_utils.py [hunk 2]
```diff
@@ -93,3 +110,498 @@ def token_counter(text):
...
...ars]ree.\'\nE       \nE       Diff is 857 characters long. Set self.maxDiff to None to see it.\n\ntwelve_days_test.py:165: AssertionError\n=========================== short test summary info ============================\nFAILED twelve_days_test.py::TwelveDaysTest::test_eighth_day_eight_maids_a_milking\nFAILED twelve_days_test.py::TwelveDaysTest::test_eleventh_day_eleven_pipers_piping\nFAILED twelve_days_test.py::TwelveDaysTest::test_fifth_day_five_gold_rings - ...\nFAILED twelve_days_test.py::TwelveDaysTest::test_fourth_day_four_calling_birds\nFAILED twelve_days_test.py::TwelveDaysTest::test_ninth_day_nine_ladies_dancing\nFAILED twelve_days_test.py::TwelveDaysTest::test_second_day_two_turtle_doves\nFAILED twelve_days_test.py::TwelveDaysTest::test_seventh_day_seven_swans_a_swimming\nFAILED twelve_days_test.py::TwelveDaysTest::test_sixth_day_six_geese_a_laying\nFAILED twelve_days_test.py::TwelveDaysTest::test_tenth_day_ten_lords_a_leaping\nFAILED twelve_days_test.py::TwelveDaysTest::test_third_day_three_french_hens\nFAILED twelve_days_test.py::TwelveDaysTest::test_twelfth_day_twelve_drummers_drumming\n========================= 11 failed, 4 passed in 0.05s =========================\n'
```
Comment (chisingh) [inline froggy/tests/test_utils.py]: These strings can be a lot shorter (... collected 25 items).

File: froggy/tests/test_utils.py [hunk 3]
```diff
@@ -93,3 +110,498 @@ def token_counter(text):
...
+
+    assert extract_max_score_from_pytest_output(message_1) == 1
```
Comment (chisingh) [inline froggy/tests/test_utils.py:470]: Also, test for 0 collected items, in which case the default return value seems to be 1.0

File: froggy/tests/test_utils.py [hunk 4]
```diff
@@ -93,3 +110,498 @@ def token_counter(text):
...
...ars]ree.\'\nE       \nE       Diff is 857 characters long. Set self.maxDiff to None to see it.\n\ntwelve_days_test.py:165: AssertionError\n=========================== short test summary info ============================\nFAILED twelve_days_test.py::TwelveDaysTest::test_eighth_day_eight_maids_a_milking\nFAILED twelve_days_test.py::TwelveDaysTest::test_eleventh_day_eleven_pipers_piping\nFAILED twelve_days_test.py::TwelveDaysTest::test_fifth_day_five_gold_rings - ...\nFAILED twelve_days_test.py::TwelveDaysTest::test_fourth_day_four_calling_birds\nFAILED twelve_days_test.py::TwelveDaysTest::test_ninth_day_nine_ladies_dancing\nFAILED twelve_days_test.py::TwelveDaysTest::test_second_day_two_turtle_doves\nFAILED twelve_days_test.py::TwelveDaysTest::test_seventh_day_seven_swans_a_swimming\nFAILED twelve_days_test.py::TwelveDaysTest::test_sixth_day_six_geese_a_laying\nFAILED twelve_days_test.py::TwelveDaysTest::test_tenth_day_ten_lords_a_leaping\nFAILED twelve_days_test.py::TwelveDaysTest::test_third_day_three_french_hens\nFAILED twelve_days_test.py::TwelveDaysTest::test_twelfth_day_twelve_drummers_drumming\n========================= 11 failed, 4 passed in 0.05s =========================\n'
```
Comment (chisingh) [inline froggy/tests/test_utils.py]: May be just keep the last line of this string. 
========================= 11 failed, 4 passed in 0.05s =========================

File: froggy/tests/test_utils.py (hunk 1 above)
Comment (xingdi-eric-yuan) [reply froggy/tests/test_utils.py:391]: I'm not super familiar with this, can you push this to my branch?

File: froggy/tests/test_utils.py (hunk 2 above)
Comment (xingdi-eric-yuan) [reply froggy/tests/test_utils.py]: done

File: froggy/tests/test_utils.py (hunk 3 above)
Comment (xingdi-eric-yuan) [reply froggy/tests/test_utils.py:470]: done

File: froggy/tests/test_utils.py (hunk 4 above)
Comment (xingdi-eric-yuan) [reply froggy/tests/test_utils.py]: done

**Pull request merged**